data/sql_setup/prepare_schedule_a.sql

```diff
@@ -124,16 +328,100 @@ $$ language plpgsql;
+    on disclosure.f_item_receipt_or_exp for each row execute procedure ofec_sched_a_delete_update(:START_YEAR_AGGREGATE);
+
+CREATE OR REPLACE FUNCTION increment_aggregates(view_row fec_fitem_sched_a_vw) RETURNS VOID AS $$
+BEGIN
+    IF view_row.contb_receipt_amt IS NOT NULL AND is_individual(view_row.contb_receipt_amt, view_row.receipt_tp, view_row.line_num, view_row.memo_cd, view_row.memo_text, view_row.contbr_id, view_row.cmte_id) THEN
+        INSERT INTO ofec_sched_a_aggregate_employer
+            (cmte_id, cycle, employer, total, count)
+        VALUES
+            (view_row.cmte_id, view_row.election_cycle, view_row.contbr_employer, view_row.contb_receipt_amt, 1)
+        ON CONFLICT ON CONSTRAINT uq_cmte_id_cycle_employer DO UPDATE
+        SET
+            total = ofec_sched_a_aggregate_employer.total + excluded.total,
+            count = ofec_sched_a_aggregate_employer.count + excluded.count;
+        INSERT INTO ofec_sched_a_aggregate_occupation
+            (cmte_id, cycle, occupation, total, count)
+        VALUES
+            (view_row.cmte_id, view_row.election_cycle, view_row.contbr_occupation, view_row.contb_receipt_amt, 1)
+        ON CONFLICT ON CONSTRAINT uq_cmte_id_cycle_occupation DO UPDATE
+        SET
+            total = ofec_sched_a_aggregate_occupation.total + excluded.total,
+            count = ofec_sched_a_aggregate_occupation.count + excluded.count;
+        INSERT INTO ofec_sched_a_aggregate_state
+            (cmte_id, cycle, state, state_full, total, count)
+        VALUES
+            (view_row.cmte_id, view_row.election_cycle, view_row.contbr_st, expand_state(view_row.contbr_st), view_row.contb_receipt_amt, 1)
+        ON CONFLICT ON CONSTRAINT uq_cmte_id_cycle_state DO UPDATE
+        SET
+            total = ofec_sched_a_aggregate_state.total + excluded.total,
+            count = ofec_sched_a_aggregate_state.count + excluded.count;
+        INSERT INTO ofec_sched_a_aggregate_zip
+            (cmte_id, cycle, zip, state, state_full, total, count)
+        VALUES
+            (view_row.cmte_id, view_row.election_cycle, view_row.contbr_zip, view_row.contbr_st, expand_state(view_row.contbr_st), view_row.contb_receipt_amt, 1)
+        ON CONFLICT ON CONSTRAINT uq_cmte_id_cycle_zip DO UPDATE
+        SET
+            total = ofec_sched_a_aggregate_zip.total + excluded.total,
+            count = ofec_sched_a_aggregate_zip.count + excluded.count;
+        INSERT INTO ofec_sched_a_aggregate_size
+            (cmte_id, cycle, size, total, count)
+        VALUES
+            (view_row.cmte_id, view_row.election_cycle, contribution_size(view_row.contb_receipt_amt), view_row.contb_receipt_amt, 1)
+        ON CONFLICT ON CONSTRAINT uq_cmte_id_cycle_size DO UPDATE
+        SET
+            total = ofec_sched_a_aggregate_size.total + excluded.total,
+            count = ofec_sched_a_aggregate_size.count + excluded.count;
```

Comment:
These calculations are all equivalent to the older table construction replaced from above, correct?

Again, nicely done with the `ON CONFLICT` clause additions! :-)